src/compile/data/data.ts

```diff
@@ -78,6 +72,12 @@ export interface AggregateComponent {
 
   /** dictionary mapping field name to string set of aggregate ops */
   measures: Dict<StringSet>;
+
+  /**
+   * Whether we need the unaggrgeated data source.
+   * Should be set when we parse the scales.
+   */
+  assembleRaw?: boolean;
```

Comment:
why should `assembleRaw` be a part of `AggregateComponent`, not `SourceComponent`?